index.js

```diff
@@ -127,7 +136,16 @@ class Ky {
 					this._response = this._fetch();
 				}
 
-				const response = await this._response;
+				let response = await this._response;
+
+				for (const hook of this._hooks.afterResponse) {
+					// eslint-disable-next-line no-await-in-loop
+					const modifiedResponse = await hook(response.clone());
+
+					if (modifiedResponse instanceof self.Response) {
+						response = modifiedResponse;
+					}
```

Comment:
This is how it's done in Got: https://github.com/sindresorhus/got/pull/640/files

Reply:
> Also, if the returned type is plain object, it should make a new try.

@szmarczak Why should it make a new try and what's special about it being a plain object?

Reply:
Options are plain object. If a plain object is returned from afterResponse hook, it is interpreted as new options (only changes present) and it makes a new try (that's how it's done in Got). 

As requested on the issue:

> I'd also love to see this feature to be able to refresh access tokens on 401 responses.

Thanks to that we can retry on 401 :) It allows users to retry whenever they want.

Reply:
> I think it should throw if the returned type is not a Response.

👍 

> Options are plain object. If a plain object is returned from afterResponse hook, it is interpreted as new options (only changes present) and it makes a new try (that's how it's done in Got).

I don't think that's a good approach. Doing completely different things based on slightly different return is not an optimal API design. It should only accept a `Response` or undefined.

For you use-case of refreshing access token, the user can just do a new request in the hook and then return the new `Response` object instead.

Reply:
> For you use-case of refreshing access token, the user can just do a new request in the hook and then return the new Response object instead.

Yeah, I didn't think about it, sorry. I gotta better change it :D

Reply:
It'd cause an infinite loop. You're calling got from got. The chain breaks. Retry count starts from 0 etc... Any better ideas?

Reply:
Maybe something like:

```js
afterResponse: [
	(response, retry) => {
		if (needsToUpdate) {
			retry(updatedOptions);
			// returns undefined
		}

		return response;
	}
]
```

I've got no other idea :|

Reply:
@szmarczak Yes, that's much better. That's an explicit action. I think we need a better name than `retry` though. This is too long, but something like `redoRequestWithChangedOptions()`. And I'm not sure why you're wrapping it in try/catch?

Reply:
> And I'm not sure why you're wrapping it in try/catch?

No idea. I wasn't fully awake yet when I was writing it :P

Reply:
Alright, so let's go with:

```js
afterResponse: [
	(response, retryWithMergedOptions) => {
		if (needsToUpdate) {
			return retryWithMergedOptions(updatedOptions);
		}

		return response;
	}
]
```

I think this can be implemented later on though. So don't worry about it, @poppinlp.